topography_generator/utils/contours.hpp

```diff
@@ -90,7 +92,14 @@ void SimplifyContours(int simplificationZoom, Contours<ValueType> & contours)
       std::vector<m2::PointD> contourSimple;
       feature::SimplifyPoints(m2::SquaredDistanceFromSegmentToPoint(),
                               simplificationZoom, contour, contourSimple);
-      contour = std::move(contourSimple);
+      // Discard closed lines which are degenerate (<=3 points) or too small for the requested zoom level.
+      /// @todo it doesn't fix all cases as the simplification algo
+      /// can produce e.g. a 5 points closed but degenerate line ("flat", not a loop anymore).
+      if (contourSimple.size() > 1 && contourSimple.front() == contourSimple.back() &&
```

Comment:
Can we put CHECK_GREATER(contourSimple.size(), 1, ()); here?

Reply:
nope, SimplifyPoints() can return an empty contourSimple

Reply:
Hm, SimplifyPoints function always keep at least 2 points if we have valid input (>= 2 points)

Reply:
So if you want to terminate on invalid input then I'll change to CHECK()

Reply:
At least we ensure that GenerateContours makes a good output.

Reply:
Ok, I've changed it